firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -520,7 +509,7 @@ private PersistedInstallationEntry fetchAuthTokenFromServer(
    * storage.
    */
   private Void deleteFirebaseInstallationId() throws FirebaseInstallationsException, IOException {
-    PersistedInstallationEntry entry = persistedInstallation.readPersistedInstallationEntryValue();
+    PersistedInstallationEntry entry = getPrefsWithGeneratedIdMultiProcessSafe();
```

Comment:
This will create a new fid and delete it. I think it is better the way it was.

Reply:
Reason for this: If there is one thread that is triggering getId or getToken and another triggering delete. If the delete read is not multi-process safe, and tries to read the same data locked by other thread acquiring CrossProcessLock - leads to OverLappingFileLockException

Reply:
As discussed offline: created and used cross-process safe read prefs method.